### PR TITLE
Support plugin event

### PR DESCRIPTION
**Pull request by ben181231**
connect #79

### CODE REVIEW

**Comment by oursky-travis** [comment]
@ben181231, thanks for your PR! By analyzing the annotation information on this pull request, we identified @rickmak to be a potential reviewer

File: skygear/transmitter/http.py [hunk 1]
```diff
@@ -74,16 +73,17 @@ def _dispatch(self, request):
         """
         kind, name, ctx, param = self.read_request(request)
         if kind == 'init':
-            skyconfig.parse_config(param.get('config', {}))
-            return self.init_info()
+            raise DeprecationWarning('Init trigger is deprecated, '
```
Comment (cheungpat) [inline skygear/transmitter/http.py]: https://docs.python.org/3.5/library/warnings.html#warning-categories:

> Base category for warnings about deprecated features (ignored by default).

So you actually need to keep the existing behaviour if you want to raise `DeprecationWarning`.

File: skygear/registry.py [hunk 2]
```diff
@@ -79,19 +81,28 @@ def register_hook(self, name, func, *args, **kwargs):
         kwargs['name'] = name
 
         if name in self.func_map['hook']:
-            log.warn("Replacing previously registered hook '%s'.", name)
+            log.warning("Replacing previously registered hook '%s'.", name)
```
Comment (cheungpat) [inline skygear/registry.py]: 👍

File: skygear/transmitter/http.py (hunk 1 above)
Comment (ben181231) [inline skygear/transmitter/http.py]: It means that `DeprecationWarning` will be ignored by default?

Comment (cheungpat) [inline skygear/transmitter/http.py]: It seems I have misunderstood what the doc says. Basically the above information only applies when you are using the `DeprecationWarning` with the `warnings` module.

I would suggest you raise other exceptions as you are actually raising an error.

Comment (ben181231) [inline skygear/transmitter/http.py]: Okay. I will update it.

**Comment by ben181231** [comment]
@cheungpat Updated.

**Pull request merged**